prime-router/src/main/kotlin/fhirengine/translation/hl7/schema/ConfigSchema.kt

```diff
@@ -0,0 +1,180 @@
+                    )
+            }
+        }
+
+        // Validate the schema elements.
+        if (elements.isEmpty())
+            addError("Schema elements cannot be empty")
+        elements.forEach { element ->
+            element.validate().forEach { addError(it) }
+        }
+
+        isValidated = true
+        return validationErrors.toList()
+    }
+}
+
+/**
+ * An element within a Schema.
+ * @property name the name of the element
+ * @property condition a FHIR path condition to evaluate. If false then the element is ignored.
+ * @property conditionExpression the validated FHIR path representation of the [condition] string
+ * @property required true if the element must have a value
+ * @property schema the name of a child schema
+ * @property schemaRef the reference to the loaded child schema
+ * @property resource a FHIR path that points to a FHIR resource
+ * @property resourceExpression the validated FHIR path representation of the [resource] string
+ * @property value a FHIR path that points to a FHIR primitive value
+ * @property valueExpression the validated FHIR path representation of the [value] string
+ * @property hl7Spec an array of hl7Specs that denote the field to place a value into
+ */
+@JsonIgnoreProperties
+data class ConfigSchemaElement(
+    var name: String? = null,
+    var condition: String? = null,
+    var conditionExpression: ExpressionNode? = null,
+    var required: Boolean? = false,
+    var schema: String? = null,
+    var schemaRef: ConfigSchema? = null,
+    var resource: String? = null,
+    var resourceExpression: ExpressionNode? = null,
+    var value: String? = null,
+    var valueExpression: ExpressionNode? = null,
+    var hl7Spec: List<String> = emptyList()
+) {
+    /**
+     * Validate the element.
+     * @return a list of validation errors, or an empty list if no errors
+     */
+    internal fun validate(): List<String> {
+        val validationErrors = mutableListOf<String>()
+
+        /**
+         * Add an error [msg] to the list of errors.
+         */
+        fun addError(msg: String) {
+            validationErrors.add("[$name]: $msg")
+        }
+
+        /**
+         * Validate and get a FHIR [path].
+         * @return the validated FHIR path.
+         */
+        fun getFhirPath(path: String?): ExpressionNode? {
+            return if (path == null) null
+            else {
+                try {
+                    FhirPathUtils.parsePath(path)
+                } catch (e: Exception) {
+                    addError("Error parsing FHIR Path")
+                    null
+                }
+            }
+        }
+
+        if (name.isNullOrBlank()) {
+            addError("Schema name cannot be blank")
+        }
+
+        // Hl7spec and value cannot be used with schema.
+        when {
+            !schema.isNullOrBlank() && (hl7Spec.isNotEmpty() || !value.isNullOrBlank()) ->
+                addError("Schema property cannot be used with hl7Spec or value properties")
```

Comment:
From here onwards there's a lot of references to `schema.isNullOrBlank()`. Is there a way they can be consolidated a little?

Reply:
I looked, but the when has two different checks: one to check is the schema is not null or blank, and the other the opposite.